hadoop-hdds/config/src/main/java/org/apache/hadoop/hdds/conf/StorageSize.java

```diff
@@ -91,6 +91,19 @@ public static StorageSize parse(String value) {
 
   }
 
+  public static StorageSize parse(String value, StorageUnit defaultUnit) {
+    try {
+      return parse(value);
+    } catch (IllegalArgumentException e) {
+      try {
+        return parse(value + defaultUnit.getShortName());
```

Comment:
Shouldn't this be
```
        return new StorageSize(defaultUnit, Double.parseDouble(value));
```

else there is redundant parsing for this unit inside the parse(..) & moreover I belive with this approach, 25K, 50M will also work because we suffix B in the end & it will become 25KB & 50MB

Reply:
Thank you @ayushtkn for the review, I have incorporated the review comments and modified the existing test cases to incorporate the newer changes.